### PR TITLE
Zstd compression

### LINKED ISSUE #11: Compression
En tant que développeur, je veux que les données soient compressées afin de sauvegarder la bande passante des utilisateurs.

Valeur: 1

### PR DESCRIPTION
**Pull request by PierreCorentin**
Closes #11

### CODE REVIEW

**Review by raffaelmoraisin — DISMISSED** [review]

File: ORA.Core.Tests/Compression/CompressionTests.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+                                                      "The dark side of the Force is a pathway to many abilities some consider to be unnatural. "+
+                                                      "He became so powerful… the only thing he was afraid of was losing his power, which eventually, "+
+                                                      "of course, he did. Unfortunately, he taught his apprentice everything he knew, then his apprentice "+
+                                                      "killed him in his sleep. Ironic. He could save others from death, but not himself.");
+
+        [Fact]
+        public void AssertNotEqualDeflate()
+        {
+            ICompressor c = Ora.GetCompressor();
+            (c.Compress(this.test)).Should().NotEqual(this.test);
```
Comment (TrAyZeN) [inline ORA.Core.Tests/Compression/CompressionTests.cs]: Remove parenthese

File: ORA.Core/Compression/ZipLibCompressor.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+using System.IO;
+using ORA.API.Compression;
+using ICSharpCode.SharpZipLib.Zip.Compression;
+
+namespace ORA.Core.Compression
+{
+    public class ZipLibCompressor : ICompressor
+    {
+
+        private static int n;
```
Comment (TrAyZeN) [inline ORA.Core/Compression/ZipLibCompressor.cs]: Not working if you compress two data and you want to decrompress the first compressed data.

**Review by TrAyZeN — CHANGES_REQUESTED** [review]

File: ORA.Core/Compression/ZipLibCompressor.cs (hunk 2 above)
Comment (adamaq01) [inline ORA.Core/Compression/ZipLibCompressor.cs]: Remove this static field

File: ORA.Core/Compression/ZipLibCompressor.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+        public byte[] Compress(byte[] data, int level)
+        {
+            n = data.Length;
```
Comment (adamaq01) [inline ORA.Core/Compression/ZipLibCompressor.cs]: Remove this assignation

File: ORA.Core/Compression/ZipLibCompressor.cs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+            {
+                int count = compressor.Deflate(buf);
+                bos.Write(buf, 0, count);
+            }
+            return bos.ToArray();
+        }
+
+        public byte[] Decompress(byte[] data)
+        {
+            byte[] result = new byte[n];
```
Comment (adamaq01) [inline ORA.Core/Compression/ZipLibCompressor.cs]: Should use a buffer and a MemoryStream

**Review by adamaq01 — CHANGES_REQUESTED** [review]
Some things may not work, changes are requested :)

**Review by TrAyZeN — DISMISSED** [review]

File: ORA.Core/Compression/ZipLibCompressor.cs [hunk 5]
```diff
@@ -29,15 +25,16 @@ public byte[] Compress(byte[] data, int level)
...
         public byte[] Decompress(byte[] data)
         {
-            byte[] result = new byte[n];
-            Inflater inf = new Inflater();
+            var bos = new MemoryStream(data.Length);
+            var inf = new Inflater();
             inf.SetInput(data);
-            int error = inf.Inflate(result, 0, n);
-            if (error == 0)
+            byte[] result = new byte[1024];
```
Comment (adamaq01) [inline ORA.Core/Compression/ZipLibCompressor.cs]: Rename to buf

**Review by adamaq01 — CHANGES_REQUESTED** [review]

**Review by adamaq01 — APPROVED** [review]

**Pull request merged**